domains/page.go

```diff
@@ -0,0 +1,23 @@
+package domains
+
+// ExclusiveStartKeyPage is object to use when returning database records in paged format using Exclusive Start Key paging
+type ExclusiveStartKeyPage[Content any] struct {
+	// List of records for the response
+	Content Content `json:"content"`
+	// Pagination details
+	Page ExclusiveStartKeyPageInfo `json:"page"`
+}
+
+// ExclusiveStartKeyPageInfo is object to use when limiting and sorting database query results for Exclusive Start Key Paging
+type ExclusiveStartKeyPageInfo struct {
+	// Field by which to order the results
+	OrderByField *string `json:"orderByField,omitempty"`
+	// Direction in which to order the results ASC/DSC
+	SortDirection *string `json:"sortDirection,omitempty"`
```

Comment:
Consider @dorzepowski 's recent cool solution for this:
![image](https://github.com/user-attachments/assets/3cbc3c73-4c93-4992-a3a6-e250de59a56a)
![image](https://github.com/user-attachments/assets/a3e7be35-7ba3-41df-a97b-f0e59264b3b0)

Reply:
What would be a benefit of this change rather than a pointer to string? Wouldn't it bascially do the same?